src/main/java/uk/gov/companieshouse/insolvency/data/api/InsolvencyApiService.java

```diff
@@ -21,33 +24,37 @@ public class InsolvencyApiService {
     private final Logger logger;
     private final String chsKafkaUrl;
     private final ApiClientService apiClientService;
+    private final ObjectMapper objectMapper;
 
     /**
      * Invoke Insolvency API.
      */
     public InsolvencyApiService(@Value("${chs.kafka.api.endpoint}") String chsKafkaUrl,
-                                ApiClientService apiClientService, Logger logger) {
+                                ApiClientService apiClientService, ObjectMapper objectMapper,
+                                Logger logger) {
         this.chsKafkaUrl = chsKafkaUrl;
         this.apiClientService = apiClientService;
+        this.objectMapper = objectMapper;
         this.logger = logger;
     }
 
     /**
      * Call chs-kafka api.
-     * @param companyNumber company insolvency number
+     * @param insolvencyDocument company insolvency document
      * @return response returned from chs-kafka api
      */
     public ApiResponse<Void> invokeChsKafkaApi(String contextId,
-                                               String companyNumber,
-                                               String eventType) {
+                                               InsolvencyDocument insolvencyDocument,
+                                               String eventType,
+                                               boolean isDelete) {
         InternalApiClient internalApiClient = apiClientService.getInternalApiClient();
         internalApiClient.setBasePath(chsKafkaUrl);
 
         PrivateChangedResourcePost changedResourcePost =
                 internalApiClient.privateChangedResourceHandler().postChangedResource(
-                        CHANGED_RESOURCE_URI,
-                        mapChangedResource(contextId, companyNumber, eventType));
-
+                        CHANGED_RESOURCE_URI, mapChangedResource(
+                                contextId, insolvencyDocument, eventType, isDelete)
+                );
```

Comment:
Instead of passing isDelete, you can check if the eventType is ("changed" or "deleted"), isDelete is not necessary